visualization_scripts/run.py

```diff
@@ -0,0 +1,145 @@
+from py2cytoscape.data.cyrest_client import CyRestClient
+from py2cytoscape.data.style import StyleUtil
+import py2cytoscape.util.cytoscapejs as cyjs
+import py2cytoscape.cytoscapejs as renderer
+from py2cytoscape import cyrest
+import networkx as nx
+import pandas as pd
+import json
+import sys
+import time
+import subprocess
+import os
+
+#assert Cytoscape is running on machine
+
+def process_exists(process_name):
+    """
+    function to assert Cytoscape is running
+
+    Args:
+        process_name (str): name of process to check for
+        example - 'Cytoscape.exe'
+
+    Returns:
+        true if process runnning, else false
+
+    """
+    bytes_name = str.encode(process_name)
+    call = 'TASKLIST', '/FI', 'imagename eq %s' % process_name
+
+    # use buildin check_output right away
+    output = subprocess.check_output(call)
+
+    # check in last line for process name
+    last_line = output.strip().split(b'\r\n')[-1]
+
+    return last_line.lower().startswith(bytes_name.lower())
+
+
+def find_path(name, path):
+    """
+    Search Algorithm to find Cytoscape application on machine
+
+    Args:
+        name (str): name of application 
+        path (str): top directory to start search from 
+
+    Returns:
+        path to first match on machine
+    """
+
+    # list to hold found matches
+    result = []
+
+    # walk directory/file tree 
+    for root, dirs, files in os.walk(path):
+        if name in files:
+            result.append(os.path.join(root, name))
+    return result[0]
+
+
+def vis(output, style):
+    """
+    REQUIRED ARGS:
+        output: output file from tps software
+        style: style file for cytoscpe 
+    """
+    
+    print("output file: " + output)
+    print("style file: " + style)
+    
+
+    #create cyrest client
+    cyy = cyrest.cyclient()
+    print("CyRest client created")
+
+    # Step 1: Create py2cytoscape client
+    cy = CyRestClient()
+
+    # Reset
+    cy.session.delete()
+    print("session reset")
+
+    # Step 2: Load network from somewhere
+    print("loading network file")
+    tps_net = cy.network.create_from(output)
+    print("Done")
+    time.sleep(2)
+
+
+    # Step 3: Apply layout
+    # list of styles to apply
+    # only contains the one style loaded from style file 
+    print("apply style file")
+    style = cyy.vizmap.load_file(style)
+    print("style name: ", style)
+    cyy.vizmap.apply(style[0])
+
+    print("Done! open up cytoscape session")
+
+
+
+def main(args):
+
+    # input args from command line
+    OUTPUT_FILE = args[1]
+    STYLE_FILE = args[2]
+    DIRNAME = r"C:\Users"
+    CYTOSCAPE = 'Cytoscape.exe'
+
+    # check if Cytoscape is running 
+    isRunning = process_exists('Cytoscape.exe')
+
+    if (isRunning == False):
+        print("Cytoscape not running")
+
+        # find path to Cytoscape on machine 
+        cytoPath = find_path(CYTOSCAPE, DIRNAME)
+
+        # open Cytoscape 
+        p = subprocess.Popen(cytoPath)
```

Comment:
I've started reading the Cytoscape installation documentation but haven't found any environment variable or other useful information that would tell us automatically where Cytoscape has been installed.  We should discuss what is most user friendly here.  Options include:
- Keep as is, user specifies Cytoscape (parent) path, script automatically launches Cytoscape if needed
- User has to open Cytoscape, we keep this check and print an error message if Cytoscape isn't detected
- We check a few common locations and try to open Cytoscape automatically if it is installed there, if not, we ask the user to launch it

Reply:
I have been playing around with some different search algorithms trying to find an optimal way to find the path to `Cytoscape.exe`. The algorithm that is in this pull request takes about 20-30 seconds to find the path to Cytoscape starting at the users directory. Could we search a few different places in parallel, and if a path is found, break out of the search and return the path?

Reply:
We decided that we will currently state that having Cytoscape installed and open is a dependency for running this script.  We will keep the code that supports searching for and launching Cytoscape but not use it by default.  It may be useful later if/when we install Cytoscape in a Docker container.

We can add comments describing that in a future version we might want to limit the runtime of the search by bounding the seconds or number of directories traversed.